### ISSUE TITLE
[question] add_option in Conan 2.0

### ISSUE THREAD

**Issue by yekmen** [conan-io/conan#15338]
### What is your question?

Hello ! 

I'm currently move our recipes to Conan 2.0. 
We use a json file where we have all 3rd library and options. 

In Conan 1.x we did something like this : 
```
    for requires in self.requires:
        _options = getOptions(requires)
        for op in _options:
            self.options[requires].add_option(op["name"], op["value"])
```
            
But in Conan 2.x add_option() function was removed. 
I tried to use 
`self.options[requires]._set()`
or
`self.options[requires].update()`

but still not working like add_option. 

Any help are welcomed. 

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @yekmen 

Thanks for your question.
It is possible in a more pythonic way (the way to dynamically set attributes):
```python
    def config_options(self):
        setattr(self.options["zlib"], "shared", True)
```
Please try that and let us know.

In any case, it is possible that there would be better recommended practices to define options in Conan 2.0, for example it is recommended to define values for dependencies in profiles better than in conanfile.py recipes.

**Comment by yekmen** [comment]
Hello @memsharded 

Sorry for the late and **happy new year** !

Thx for you help, I have something like this: 

```
def requirements(self):
        self.requires(conanhelper.getVersion("opencv", self)) #conanhelper.getVersion will return opencv/4.6.0 according our json file

def config_options(self):
        setattr(self.options["opencv"], "world", True)
```

the problem is, `config_options` is called before `requirements` I guess, self.requires.values() is empty.

**Comment by memsharded** [comment]
> Sorry for the late and happy new year !

Happy new year to you too!

> the problem is, config_options is called before requirements I guess, self.requires.values() is empty.

This shouldn't be a problem, because that ``self.options`` is decoupled from the ``requirements`` it shouldn't be an issue that it is processed earlier. Have you tried and it failed? Can you please share the error messages?

**Comment by yekmen** [comment]
I have no issue, this is more about our helper and how we do with Conan 1.x.

```
def setOptions(self):
    #---------- set options of libs --------- #
    for requires in self.requires.values():
     _options = getOptions(str(requires.ref.name), str(self.settings.os))	#get options from JSON Files
     if not _options:
        continue
     y = json.dumps(_options)
     x = json.loads(y)
     for op in x:
         setattr(self.options[requires], op["name"], op["value"])
```

I need to parse the required package and set the options. 
Maybe is not the correct workflow but is working well with Conan 1.x.

Maybe some few word about what I need and what I tring to make. 
We have couple of projects which depends together and I tried to make all dependencies the same (version + options)

I saw that something is possible with profile.

**Comment by yekmen** [comment]
I will close the issue. 

Summarize for someone who want to use add_option in Conan 2.0

**`  def config_options(self):
        setattr(self.options["zlib"], "shared", True)`**
        
is working well only if you already define your Requirement, it was not my case.
So now I changed my mind and use profiles. 
I created a **main_profile** which contains all dependencies of my main project and include it on my subprojects.

Thx @memsharded  for your support
